### PR TITLE
[fix] domain remove if an app without a domain is installed

### PR DESCRIPTION
**Pull request by kay0u**
## The problem

...

## Solution

Add small check to avoid a crash

## PR Status

...

## How to test

...

## Validation

- [ ] Principle agreement 0/2 : 
- [ ] Quick review 0/1 : 
- [ ] Simple test 0/1 : 
- [ ] Deep review 0/1 :

### CODE REVIEW

**Comment by Psycojoker** [comment]
I hate shitty error messages so much that now it's:

```
root@ynh:/vagrant/yunohost/src/yunohost# yunohost domain remove qsd.dev
Error: Those applications are still installed on your domain: my_webapp (on https://qsd.dev/site), wordpress (on https://qsd.dev/blog). Please uninstall them before proceeding to domain removal
```

While it was that before:

```
root@ynh:/vagrant/yunohost/src/yunohost# yunohost domain remove qsd.dev
Error: One or more apps are installed on this domain. Please uninstall them before proceeding to domain removal
```

**Comment by Psycojoker** [comment]
LGTM, also have tested the PR at the same time

File: src/yunohost/domain.py [hunk 1]
```diff
@@ -179,9 +179,15 @@ def domain_remove(operation_logger, domain, force=False):
...
     # Check if apps are installed on the domain
-    app_settings = [_get_app_settings(app) for app in _installed_apps()]
-    if any(s["domain"] == domain for s in app_settings):
-        raise YunohostError('domain_uninstall_app_first')
+    apps_on_that_domain = []
+
+    for app in _installed_apps():
+        settings = _get_app_settings(app)
+        if settings.get("domain") == domain:
+            apps_on_that_domain.append("%s (on https://%s%s)" % (app, domain, settings.get("path")))
```
Comment (alexAubin) [inline src/yunohost/domain.py:187]: Not all apps are webapps tho :s 

```suggestion
            apps_on_that_domain.append("%s (on https://%s%s)" % (app, domain, settings["path"]) if "path" in settings else app))
```

Comment (Psycojoker) [reply src/yunohost/domain.py:187]: Ah, this is going to do a "https://domain.tld/$app_id" so we actually even want something else :<

Comment (Psycojoker) [reply src/yunohost/domain.py:187]: Oupsi, spoke too fast

**Pull request merged**